docs/cli-arguments.md

```diff
@@ -42,12 +42,16 @@ Usage of ./kube-state-metrics:
       --metric-blacklist string                     Comma-separated list of metrics not to be enabled. This list comprises of exact metric names and/or regex patterns. The whitelist and blacklist are mutually exclusive.
       --metric-whitelist string                     Comma-separated list of metrics to be exposed. This list comprises of exact metric names and/or regex patterns. The whitelist and blacklist are mutually exclusive.
       --namespace string                            Comma-separated list of namespaces to be enabled. Defaults to ""
+      --pod string                                  Name of the pod. When set, it is expected that --pod and --pod-namespace are both set. This is used for auto-detecting sharding. If set, this has preference over statically configured sharding. This is experimental, it may be removed without notice.
+      --pod-namespace string                        Name of the namespace. When set, it is expected that --pod and --pod-namespace are both set. This is used for auto-detecting sharding. If set, this has preference over statically configured sharding. This is experimental, it may be removed without notice.
```

Comment:
Isn't it safe to assume the default pod and pod namespaces names to be - `kube-state-metrics` and `kube-system` respectively? 

If so, we can remove the requirement for both pod name and pod namespace to be set.

Reply:
We can't assume that I think and I'd prefer to have to explicitly set this instead of a default. I feel a default would be more error prone. I prefer to handle this via documented manifests.